src/core/tag/mixins/provide-consume.ts

```diff
@@ -0,0 +1,61 @@
+import Tag from '..';
+import * as utils from '../../utils';
+import Phase from '../phase';
+import TagUtils from '../utils';
+
+export default function provideConsumeMixin(this: Tag) {
+  this.one(Phase.INITIALIZE, () => {
+    const { provides = {} } = Tag.getMeta(this);
+
+    if (Object.keys(provides).length !== 0) {
```

Comment:
Is `Object.assign` expensive enough to warrant this check?

Reply:
Is this check expensive enough to warrant taking it out?